### ISSUE TITLE
Use the dependency graph format for the GoMod package manager

### ISSUE THREAD

**Issue by oheger-bosch** [oss-review-toolkit/ort#4249]
This is a sub ticket of #3825 that addresses the GoMod package manager implementation.

**Comment by sschuberth** [comment]
@fviernau, in order to move a bit forward with the #3825 epic, I was wondering whether you'd have a chance to work on this issue as you were doing a lot of Go(Mod) stuff recently?

**Comment by fviernau** [comment]
I'd actually be interested in doing this, I would volunteer but I cannot make promises about any E.T.A. right now. 
It may make sense to do it after #5555, as I expect to unlock quite a bit of simplification potential in that ticket.

**Comment by sschuberth** [comment]
> I'd actually be interested in doing this

@fviernau, is there still interested from your side to implement his?

**Comment by fviernau** [comment]
> @fviernau, is there still interested from your side to implement his?

Interest yes, available time in near future: no. 
So, I cannot make any promises on that right now, sadly.

**Comment by sschuberth** [comment]
> Not make GoMod more complicated but replace or refactor the depdendency graph API so that it better fits.

Fair enough, then we should probably involve @oheger-bosch to check whether he'd have time for such a refactoring...

**Comment by fviernau** [comment]
@schuberth what's the main reason for the migration? 
1. Consistency
2. Removal of the redundant `breakCycles()` code
3. Anything else

**Comment by sschuberth** [comment]
I don't think there *one* main reason, but a mix of
- consistency / reduction of similar code
- preparing for the removal of legacy dependency tree code (once [all PMs have been migrated](https://github.com/oss-review-toolkit/ort/issues/3825))
- the hope for reduced memory usage during analysis

**Comment by sschuberth** [comment]
Closing this in favor of tracking it only in parent issue #3825.
